server/src/leaderboard.ts

```diff
@@ -0,0 +1,328 @@
+import { Chain } from '../shovel-config'
+import db from './db'
+
+type LeaderboardLine = {
+  problemId: string
+  owner: string
+}
+
+type SpeedScore = LeaderboardLine & { ticks: number }
+
+type SolvedScore = LeaderboardLine & { solved: number }
+
+type StreakScore = LeaderboardLine & { streak: number }
+
+type DailyLeaderboard = {
+  oneBody: SpeedScore[]
+  twoBody: SpeedScore[]
+  threeBody: SpeedScore[]
+  cumulative: SpeedScore[]
+}
+
+type Leaderboard = {
+  daily: Record<number, DailyLeaderboard>
+  allTime: {
+    mostSolved: SolvedScore[]
+    currentStreak: StreakScore[]
+    fastest: SpeedScore[]
+  }
+}
+
+export const leaderboard: Leaderboard = {
+  daily: {},
+  allTime: {
+    mostSolved: [],
+    currentStreak: [],
+    fastest: []
+  }
+}
+
+function beginningOfTodayInUTCSeconds(): number {
+  const now = new Date()
+  const startOfTodayUTC = new Date(
+    Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate())
+  )
+  return Math.floor(startOfTodayUTC.getTime() / 1000)
```

Comment:
another method confirmed to be working with the solidity implementation in case you want to use in a test
```js

    function currentDayInUnixTime() {
      const date = new Date()
      date.setUTCHours(0, 0, 0, 0)
      return date.getTime() / 1000
    }
```

Reply:
cool i will use that!